src/main/java/com/microsoft/azure/spring/data/cosmosdb/exception/CosmosDBAccessException.java

```diff
@@ -8,12 +8,39 @@
 import org.springframework.dao.DataAccessException;
 import org.springframework.lang.Nullable;
 
+/**
+ * Public class extending DataAccessException, exposes innerException.
+ * Every API in {@link com.microsoft.azure.spring.data.cosmosdb.repository.CosmosRepository}
+ * and {@link com.microsoft.azure.spring.data.cosmosdb.repository.ReactiveCosmosRepository}
+ * should throw {@link CosmosDBAccessException}.
+ * innerException refers to the exception thrown by CosmosDB SDK. Callers of repository APIs can
+ * rely on innerException for any retriable logic, or for more details on the failure of
+ * the operation.
+ */
 public class CosmosDBAccessException extends DataAccessException {
+
+    protected final Exception innerException;
```

Comment:
Maybe a more concrete exception type in cosmos sdk could be used.

Reply:
Yes, that is a valid point, we should change it to `CosmosClientException`. Cosmos SDK is supposed to throw only `CosmosClientException`, so we can expose it through here.